lib/jobs/local-catalog.js

```diff
@@ -0,0 +1,180 @@
+// Copyright 2015, EMC, Inc.
+/* jshint: node:true */
+
+'use strict';
+
+var di = require('di');
+
+module.exports = LocalCatalogJobFactory;
+di.annotate(LocalCatalogJobFactory, new di.Provide('Job.Local.Catalog'));
+    di.annotate(LocalCatalogJobFactory,
+    new di.Inject(
+        'Job.Base',
+        'JobUtils.CommandParser',
+        'Services.Waterline',
+        'Logger',
+        'Promise',
+        'Assert',
+        'Util',
+        '_',
+        'Services.Encryption',
+        'ChildProcess'
+    )
+);
+function LocalCatalogJobFactory(
+    BaseJob, 
+    parser, 
+    waterline, 
+    Logger, 
+    Promise, 
+    assert, 
+    util, 
+    _,
+    encryption,
+    ChildProcess
+) {
+    var logger = Logger.initialize(LocalCatalogJobFactory);
+
+    /**
+     * @param {Object} options
+     * @param {Object} context
+     * @param {String} taskId
+     * @constructor
+     */
+    function LocalCatalogJob(options, context, taskId) {
+        LocalCatalogJob.super_.call(this, 
+                                   logger,
+                                   options, 
+                                   context, 
+                                   taskId);
+
+        assert.string(this.context.target);
+        assert.arrayOfString(this.options.commands);
+
+        this.nodeId = this.context.target;
+        this.commands = _.isArray(options.commands) ? options.commands : [options.commands];
+    }
+    util.inherits(LocalCatalogJob, BaseJob);
+
+    /**
+     * @memberOf LocalCatalogJob
+     */
+    LocalCatalogJob.prototype._run = function() {
+        var self = this;
+
+        waterline.nodes.findByIdentifier(self.nodeId)
+        .then(function (node) {
+            assert.ok(node, 'No node for local catalog');
+            return Promise.map(self.commands, function(cmd) {
+                return parser.validateParser(cmd)
+                .then(function() {
+                    return self.runCommand(cmd);
+                })
+                .catch(function(err) {
+                    logger.error("Run error", { command: cmd, error: err });
```

Comment:
Do we want to throw here and fail the job if a command isn't in the whitelist, rather than just dropping it?